jerry-core/jerry.cpp

```diff
@@ -899,6 +900,62 @@ bool jerry_api_get_object_field_value (jerry_api_object_t *object_p,
 }
 
 /**
+ * Applies the given function to the every fields in the objects
+ *
+ * @return true. if object traversal is successfully. i.e.:
+ *                - no unhandled exceptions were thrown in traversal of object properties;
+ *         false - otherwise, any exception is thrown during the traversal. i.e:
+ *                - unhandled exception were thrown in property getter;
+ */
+bool
+jerry_api_foreach_object_field (jerry_api_object_t *object_p, /**< object */
+                               jerry_object_field_foreach_t foreach_p, /**< foreach function */
+                               void *user_data_p) /**< user data for foreach function */
+{
+  jerry_assert_api_available ();
+
+  bool continuous = true;
+  for (ecma_property_t *property_p = ecma_get_property_list (object_p);
+       property_p != NULL && continuous;
+       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  {
+    ecma_string_t *property_name_p;
+    if (property_p->type == ECMA_PROPERTY_NAMEDDATA)
+    {
+      property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t, property_p->u.named_data_property.name_p);
+    }
+    else if (property_p->type == ECMA_PROPERTY_NAMEDACCESSOR)
+    {
+      property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t, property_p->u.named_accessor_property.name_p);
+    }
+    else
+    {
+      JERRY_ASSERT (property_p->type == ECMA_PROPERTY_INTERNAL);
+      continue;
+    }
+    if (!ecma_is_property_enumerable (property_p))
+    {
+      continue;
+    }
+
+    ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+    ECMA_TRY_CATCH (property_value, ecma_op_object_get (object_p, property_name_p), ret_value);
+    jerry_api_value_t field_value;
+    jerry_api_convert_ecma_value_to_api_value (&field_value, property_value);
+    continuous = foreach_p (property_name_p, &field_value, user_data_p);
+    jerry_api_release_value (&field_value);
+    ECMA_FINALIZE (property_value);
+    if (ecma_is_completion_value_throw (ret_value))
```

Comment:
I would suggest using the `!ecma_is_completion_value_empty (ret_value)` condition instead. Also please add an empty line before the if.